opengever/maintenance/scripts/repository_migration_analyse.py

```diff
@@ -0,0 +1,468 @@
+        ann = IAnnotations(transmogrifier)
+        ann[BUNDLE_PATH_KEY] = bundle_path
+        ann[INTERMEDIATE_COMMITS_KEY] = False
+
+        with DisabledLDAP(portal):
+            transmogrifier(u'opengever.bundle.oggbundle')
+
+    def move_branches(self, items):
+        mapping = self.get_repository_reference_mapping()
+        for item in items:
+            parent = uuidToObject(item['new_parent_uid'])
+            if not parent:
+                parent = self.catalog(guid=item['new_parent_uid'])[0].getObject()
+
+            repo = uuidToObject(item['uid'])
+            if not parent or not repo:
+                raise Exception('No parent or repo found for {}'.format(item))
+
+            api.content.move(source=repo, target=parent, safe_id=True)
+
+    def get_repository_reference_mapping(self):
+        if not self._reference_repository_mapping:
+            repos = [brain.getObject() for brain in
+                     self.catalog(object_provides=IRepositoryFolder.__identifier__)]
+            self._reference_repository_mapping = {
+                repo.get_repository_number(): repo for repo in repos}
+
+        return self._reference_repository_mapping
+
+    def adjust_reference_number_prefix(self, items):
+        parents = set()
+        for item in items:
+            repo = uuidToObject(item['uid'])
+            referenceprefix.IReferenceNumberPrefix(repo).reference_number_prefix = item['new_number']
+            parents.add(aq_parent(aq_inner(repo)))
+
+        self.regenerate_reference_number_mapping(list(parents))
+
+    def regenerate_reference_number_mapping(self, objs):
+        for obj in objs:
+            ref_adapter = IReferenceNumberPrefix(obj)
+            # This purges also the dossier mapping, but the parents does not
+            # contain any dossier otherwise something is wrong and an
+            # exception will be raised when looping over the childs.
+            ref_adapter.purge_mappings()
+
+            for child in obj.listFolderContents():
+                if not IRepositoryFolder.providedBy(child):
+                    raise Exception(
+                        'A parent of a repositoryfolder contains dossiers')
+                ref_adapter.set_number(
+                    child, number=child.reference_number_prefix)
+
+    def rename(self, items):
+        for item in items:
+            repo = uuidToObject(item['uid'])
+
+            # Rename
+            repo.title_de = item['new_title']
+
+            # Adjust id if necessary
+            ObjectIDUpdater(repo, FakeOptions()).maybe_update_id()
+
+    def update_description(self, items):
+        for item in items:
+            repo = uuidToObject(item['uid'])
+            if not repo:
+                continue
+
+            new_description = item['new_item']['description']
+            if repo.description != new_description:
+                repo.description = new_description
+
+    def reindex(self):
```

Comment:
Shouldn't we also reindex all children of moved objects? Maybe that happened automatically with the move